OpenRA.Game/Actor.cs

```diff
@@ -148,9 +175,34 @@ internal void Created()
 		{
 			created = true;
 
+			// Make sure traits are usable for condition notifiers
 			foreach (var t in TraitsImplementing<INotifyCreated>())
 				t.Created(this);
 
+			var allObserverNotifiers = new HashSet<VariableObserverNotifier>();
+			foreach (var provider in TraitsImplementing<IObservesVariables>())
+			{
+				foreach (var variableUser in provider.GetVariableObservers())
+				{
+					allObserverNotifiers.Add(variableUser.Notifier);
+					foreach (var variable in variableUser.Variables)
+					{
+						var cs = conditionStates.GetOrAdd(variable);
+						cs.Notifiers.Add(variableUser.Notifier);
+
+						// Make sure the variable is set.
```

Comment:
Can we please change this comment to something like
```csharp
// Initialize conditions that have not yet been granted to 0
// NOTE: Some conditions may have already been granted by INotifyCreated calling GrantCondition,
// and we choose to assign the token count to safely cover both cases instead of adding an if branch.
```
to make the situation (at least, this is my understanding of it) clearer?

Reply:
Comment updated.